### PR TITLE
[AIRFLOW-6562] mushroom cloud error when clicking 'mark failed/success' from graph view of dag that has never been run yet

### PR DESCRIPTION
**Pull request by tooptoop4**
mushroom cloud error when clicking 'mark failed/success' from graph view of dag that has never been run yet

Issue link: [AIRFLOW-6562](https://issues.apache.org/jira/browse/AIRFLOW-6562)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/api/common/experimental/mark_tasks.py [hunk 1]
```diff
@@ -102,6 +102,10 @@ def set_state(
     if dag is None:
         raise ValueError("Received tasks with no DAG")
 
+    latest_execution_date = dag.latest_execution_date
+    if latest_execution_date is None:
+        return []
+
```
Comment (xinbinhuang) [inline airflow/api/common/experimental/mark_tasks.py:108]: This whole snippet should be modified inside `get_execution_dates`. 
https://github.com/apache/airflow/blob/2eff0e750f5a0cfde6bb68af215631f781e8fd90/airflow/api/common/experimental/mark_tasks.py#L246-L250

Also, isn't the original behavior (throwing an error) intended?

Comment (tooptoop4) [reply airflow/api/common/experimental/mark_tasks.py:108]: user is doing simple click from UI, nothing selected so why should they see nasty error?

Comment (xinbinhuang) [reply airflow/api/common/experimental/mark_tasks.py:108]: Double-checked this. I agree that it should not throw an (internal?) error. I am wondering what's the reason for this error handling @jmcarp @ashb @kaxil @potiuk as this is only used by the frontend `airflow/www/views.py`.

Comment (ashb) [reply airflow/api/common/experimental/mark_tasks.py:108]: That error message is a bit of a misnomer. It used to look like this:

```diff
-    assert latest_execution_date is not None
+    if latest_execution_date is None:
+        raise ValueError("Received non-localized date {}".format(execution_date))
```

I agree with @xinbinhuang - this change should go in `get_execution_dates` (and check the value of `dates` after the call to short circuit the rest of the function.

**Review by ashb — CHANGES_REQUESTED** [review]

**Comment by tooptoop4** [comment]
closing as will be covered by https://github.com/apache/airflow/pull/7709

**Pull request closed without merging**